### PR TITLE
Download page: open the OS specific tab on page load

### LINKED ISSUE #1115: Download page: open the OS specific tab on page load
**Introduce your stance**

MacOS users generally want to install Nix on their own system. I can't exclude the possibility that they want to install it on Ubuntu, but that's what the tab menu is for.

**Describe the issue**

When opening the download page on Safari, it shows the Linux instructions.

This comes across as inattentive, even if it's not a huge problem.

**Page links**

https://nixos.org/download

**Additional context**

Probably there's a more clever way to check for macOS rather than just Safari.

### PR DESCRIPTION
**Pull request by VedantWankhade**
Closes #1115 
Uses `navigator.userAgent` to find user's platform and sets the initial active tab accordingly.

### CODE REVIEW

File: src/components/pages/download/DownloadAccordion.astro [hunk 1]
```diff
@@ -67,6 +67,23 @@ let downloadOptions = (await getCollection("download")).filter(
...
+      os = "macos";
+    } else if(userPlatform.includes("windows")) {
+      os = "windows";
+    }
+    for (let i = 0; i < panels.length; i++) {
+      if (panels[i].id === `nix-install-${os}`) {
+        activePanel = panels[i];
+        break;
+      }
+    }
```
Comment (thilobillerbeck) [inline src/components/pages/download/DownloadAccordion.astro:84]:
```suggestion
    activePanel = acc.querySelector(`#nix-install-${os}`);
```

**Review by thilobillerbeck — CHANGES_REQUESTED** [review]
Thanks for contributing, I have one small suggestion

File: src/components/pages/download/DownloadAccordion.astro (hunk 1 above)
Comment (VedantWankhade) [reply src/components/pages/download/DownloadAccordion.astro:84]: That's clever. Actually I tried the same approach the first time but I used `panels` to query on but obviously that didnt work and somehow I overlooked the `acc` variable.

**Comment by VedantWankhade** [comment]
I have made changes according to the suggestion.

**Review by thilobillerbeck — APPROVED** [review]
Thanks for your contribution

**Pull request merged**